Backups Dashboard: add restore page scaffolding

Initsogar opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

Part of DOTDASH-282

This PR aims to add the foundation code to start iterating on the Restore page (DOTDASH-278).

## Proposed Changes

* Add `/backups/$rewindId/restore` route to the dashboard router.
* Add restore page scaffolding.
* Refactor backup actions to use router for navigation.

<img width="1424" height="824" alt="CleanShot 2025-08-14 at 22 21 17@2x" src="https://github.com/user-attachments/assets/c4a7139f-8de9-4db6-822b-109f519215ad" />

### Demo (updated after https://github.com/Automattic/wp-calypso/pull/105166)

https://github.com/user-attachments/assets/e4d172ec-5e80-4d8f-81d6-4241d3789ba3

Old video: https://github.com/user-attachments/assets/1147caf0-9466-426d-8392-19b22940bd33

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Part of Backups in the Hosting Dashboard

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Navigate to /v2/sites/SITE_SLUG/backups using a site with the backup feature
* Click on any of the backups on the list, then click on -> Restore to this point
* It should redirect you to `/v2/sites/ragostini.com/backups/#####/restore`
* Clicking on `< Backups` should bring you back to the Backups page.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Initsogar committed:
Add site backups index and restore routes to the dashboard router

- Introduced `siteBackupsIndexRoute` for the backups index page.
- Added `siteBackupRestoreRoute` for restoring backups.
- Updated `createRouteTree` to include new routes under the backups section.

Initsogar committed:
Add restore page scaffolding

Initsogar committed:
Refactor backup actions to use router for navigation

arthur791004 commented on client/dashboard/app/router.tsx:
```diff
@@ -270,14 +270,30 @@ const siteLogsChildRoutes = [ siteLogsIndexRoute, siteLogsPhpRoute, siteLogsServ
 const siteBackupsRoute = createRoute( {
 	getParentRoute: () => siteRoute,
 	path: 'backups',
+} );
```
I think we should continue rendering `sites/backups` here so the upsell is shown under all backups-related routes.
Next, we should pass the `<Outlet />` component to the main prop of `<CalloutOverlay />`.

With this setup:
* `siteBackupsIndexRoute` will render the data views.
* `siteBackupRestoreRoute` will render the restore component.

Both routes can still remain gated under the upsell.

Initsogar replied:
Good tip :). I just addressed it in https://github.com/Automattic/wp-calypso/pull/105221/commits/618a55556525f63c38aba8928bc63bf76f7368c5.

https://github.com/user-attachments/assets/e4d172ec-5e80-4d8f-81d6-4241d3789ba3

arthur791004 commented on client/dashboard/sites/backups/restore/index.tsx:
```diff
@@ -0,0 +1,65 @@
+function SiteBackupRestore() {
+	const { siteSlug, rewindId } = siteBackupRestoreRoute.useParams();
+	const router = useRouter();
+	const { data: site } = useQuery( siteBySlugQuery( siteSlug ) );
```
We can use`useSuspenseQuery` to ensure the `site` is always available as we do the preload on the router 🙂

Initsogar replied:
Good tip. I just addressed it on https://github.com/Automattic/wp-calypso/pull/105221/commits/98e42fe93adc8d98277e1a276fa77faa77a900f1.
It wasn't needed at all in this component, but I addressed it in the others.

fushar commented on client/dashboard/sites/backups/restore/index.tsx:
```diff
@@ -0,0 +1,65 @@
+import { PageHeader } from '../../../components/page-header';
+import PageLayout from '../../../components/page-layout';
+
+function SiteBackupRestore() {
```
Can we put the directory as `sites/backup-restore` (flat) to align with existing pattern 🙂

Initsogar replied:
Thanks, addressed in https://github.com/Automattic/wp-calypso/pull/105221/commits/380c8e438b9e161fad16d6021725e7198457e336

Initsogar committed:
Merge remote-tracking branch 'origin/trunk' into add/hd-backups-restore-scaffolding

Initsogar committed:
Adjust layout of BackupDetails and add restore action

Initsogar committed:
Relocate `sites/backups/restore` to `sites/backup-restore` to align with existing pattern

Initsogar committed:
Refactor backups page to enable upsell on all backup-related routes

Initsogar committed:
Remove unused imports and query from SiteBackupRestore component

Initsogar committed:
Refactor backups page to use `useSuspenseQuery` for fetching the site by slug

Initsogar commented:
I just addressed all the comments and updated the video demo, after merging https://github.com/Automattic/wp-calypso/pull/105166. It would be great if you can take another look :) @arthur791004 @fushar 

https://github.com/user-attachments/assets/e4d172ec-5e80-4d8f-81d6-4241d3789ba3

myhro commented on client/dashboard/sites/backup-restore/index.tsx:
```diff
@@ -0,0 +1,58 @@
+	const router = useRouter();
+
+	const backButton = (
+		<Button
```
I noticed that when clicking on this button, we lose track of the backup which was previously selected in the list. Not a blocker, but we may want to track that as an issue.

Initsogar replied:
Good catch. I think we could track this by either passing the rewindId as a URL param. I'll create a task to track this.

myhro commented on client/dashboard/sites/backups/dataviews/actions.tsx:
I think we won't be using actions anymore in the list view, as they were useful as the last column in the table view. In any case, we can drop all of them in a future PR after confirming that.

Initsogar replied:
Agree! Let's address it in a separate cleanup PR.

myhro commented on client/dashboard/sites/backups/index.tsx:
```diff
@@ -56,28 +57,10 @@ export function SiteBackupsCallout( {
-}
-
-function SiteBackups() {
+export function BackupsListPage() {
```
I was a bit confused by this change, but I think we are just renaming `BackupsLayout` to `BackupsListPage`, right? Not sure if it needs to be exported.

Initsogar replied:
Yes, renaming `BackupsLayout` to `BackupsListPage` for clarity. The export is needed because it's imported in [router.tsx:300 for the backups index route](https://github.com/Automattic/wp-calypso/pull/105221/files/98e42fe93adc8d98277e1a276fa77faa77a900f1). This change allows the backups section to have nested route:  the parent component (`SiteBackups`) handles the callout overlay (upsell prompt) for sites without backups, while BackupsListPage is the specific component for the main page, which have Backups list.

myhro reviewed (approved):
Thanks for the PR, Rafa! I've made a few comments, but none of them are blockers. ✅
